### PR TITLE
Added Metrics docummentation

### LINKED ISSUE #3400: Need document for monitoring metrics
/kind enhancement
The kubevirt metrics are not documented. They should be listed in monitoring section or some where else. 
I have few questions about the metrics：
1. What does `kubevirt_vmi_phase_count` mean? 
2. Is `kubevirt_vmi_network_errors_total` including packet dropped？If not, is there any plan to support packet drop metric like `container_network_transmit_packets_dropped_total` in cadvisor? 
3. What does `state="1"` means in `kubevirt_vmi_vcpu_seconds{xxxx,state="1"}` metric? 

**Anything else we need to know?**:

**Environment**:
- KubeVirt version (use `virtctl version`):v0.28.0
- Kubernetes version (use `kubectl version`):
- VM or VMI specifications:
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by ArthurSens**
Signed-off-by: arthursens arthursens2005@gmail.com

What this PR does / why we need it:
Add a document to explain kubevirt vmi metrics

Which issue(s) this PR fixes
Fixes #3400

Special notes for your reviewer:
There is still a lot to be done. Maybe some examples could help... but know exactly which metrics should need examples e which one shouldn't

Metrics for Reflector, Client and Workqueue I think that could be docummented on another PR, since I don't have the necessary knowledge right now and could take quite some time

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by dhiller** [comment]
@ArthurSens I added the release not block which is required for [tide](https://github.com/kubernetes/test-infra/tree/master/prow/tide). See [tide configuration](https://github.com/kubevirt/project-infra/blob/master/github/ci/prow/files/config.yaml#L107) for more details.

File: docs/metrics.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+| id | Indentifier to a single Virtual CPU. |
+| state | Identify the Virtual CPU state. It can be one of libvirtd vcpu's states: `OFFLINE`, `RUNNING` or `BLOCKED` |
+
+#### kubevirt_vmi_phase_count
+
+Total amount of VMIs per node and phase.
+
+| Label | Description |
+|-------|-----------------------------------------------------------------------------------------------------------------------------------------|
+| phase | Phase of the VMI. It can be one of [kubernetes pod lifecycle phases](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/) |
```
Comment (xpivarc) [inline docs/metrics.md:135]: I would rather refer to https://github.com/kubevirt/kubevirt/blob/master/staging/src/kubevirt.io/client-go/api/v1/types.go#L415.

File: docs/metrics.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+#### kubevirt_vmi_memory_resident_bytes
+
+Total resident memory of the process running the VMI. Usually set on VMI manifest's [like this example](../examples/vmi-ephemeral.yaml#L8-19)
+
+| Label | Description |
+|-----------|----------------------------------------------|
+| name | VMI's name given on it's specification. |
+| namespace | Namespace which the given VMI is related to. |
+| node | Node where the VMI is running on. |
+
```
[the remark this answers is not in the document]
Comment (ArthurSens) [reply docs/metrics.md:23]: We do have [a issue to remove domain labels](https://github.com/kubevirt/kubevirt/issues/3477), and a [PR as well](https://github.com/kubevirt/kubevirt/pull/3500)

Should I add these labels even if it's going to be removed soon?

[the remark this answers is not in the document]
Comment (xpivarc) [reply docs/metrics.md:23]: Ah, I see. It should be fine as long as #3500 will be not missed/forgotten.

**Comment by xpivarc** [comment]
/lgtm
As discussed, we will add some autogeneration but this looks good to me.

**Pull request merged**